### PR TITLE
MOTECH-2908 Added Encounter UUID as post-action field

### PR DESCRIPTION
**Pull request by jbanasiewicz**
Added Encounter UUID as post-action field, changed createEncounter method from "void" to "Encounter" type.

### CODE REVIEW

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

File: openmrs/src/main/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceImpl.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public void createEncounter(String configName, DateTime encounterDatetime, Strin
         EncounterType type = new EncounterType(encounterType);
 
         Encounter encounter = new Encounter(location, type, encounterDatetime.toDate(), patient, Collections.singletonList(provider.getPerson()), observationList);
-        encounterService.createEncounter(configName, encounter);
+        return encounterService.createEncounter(configName, encounter);
```
Comment (kmadej) [inline openmrs/src/main/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceImpl.java:78]: adjust unit test to verify if this method returns valid encounter object

**Comment by kmadej** [comment]
test this please

**Comment by kmadej** [comment]
please add integration test in a separate PR

File: openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java [hunk 2]
```diff
@@ -134,6 +135,9 @@ public void shouldCreateEncounterWithGivenParameters() {
         verify(encounterService).createEncounter(eq(CONFIG_NAME), encounterCaptor.capture());
 
         assertEquals(encounter, encounterCaptor.getValue());
```
Comment (kmadej) [inline openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java]: this equal is now redundant, just check if action returns proper encounter object

File: openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java [hunk 3]
```diff
@@ -134,6 +135,9 @@ public void shouldCreateEncounterWithGivenParameters() {
...
+        assertEquals(encounter, openMRSActionProxyService.createEncounter(CONFIG_NAME, new DateTime(encounter.getEncounterDatetime()),
```
Comment (kmadej) [inline openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java]: you can use encounter object from previous createEncounter invocation

File: openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java [hunk 4]
```diff
@@ -124,16 +124,17 @@ public void shouldCreateEncounterWithGivenParameters() {
...
+        doReturn(encounter).when(encounterService).createEncounter(eq(CONFIG_NAME), eq(encounter));
         doReturn(Collections.singletonList(location))
                 .when(locationService).getLocations(eq(CONFIG_NAME), eq(location.getName()));
 
-        openMRSActionProxyService.createEncounter(CONFIG_NAME, new DateTime(encounter.getEncounterDatetime()),
+        Encounter encounterCreated = openMRSActionProxyService.createEncounter(CONFIG_NAME, new DateTime(encounter.getEncounterDatetime()),
                 encounter.getEncounterType().getName(), location.getName(), patient.getUuid(), provider.getUuid(),
                 observations);
 
         verify(encounterService).createEncounter(eq(CONFIG_NAME), encounterCaptor.capture());
```
Comment (kmadej) [inline openmrs/src/test/java/org/motechproject/openmrs/tasks/impl/OpenMRSActionProxyServiceTest.java]: remove this line

**Comment by kmadej** [comment]
test this please

**Comment by kmadej** [comment]
test this please

**Comment by kmadej** [comment]
test this please

**Comment by kmadej** [comment]
test this please

**Pull request merged**